File: src/main/kotlin/de/brainsizzle/twitchbotpaint/paint/RotateAnimation.java
package de.brainsizzle.twitchbotpaint.paint;

public class RotateAnimation implements Animation {

    double degreesToRotatePerStep;
    int stepsRemaining;

    public RotateAnimation(double degreesToRotatePerStep, int stepsRemaining) {
        this.degreesToRotatePerStep = degreesToRotatePerStep;
        this.stepsRemaining = stepsRemaining;
    }

    @Override
    public boolean animate(Shape shape) {
        System.out.println("rotating " + stepsRemaining + " " + degreesToRotatePerStep);
        if (stepsRemaining > 0) {
            shape.setRotationDegress(shape.getRotationDegress() + degreesToRotatePerStep);
            stepsRemaining--;
            return false;
        }
        return true;
    }

}

File: src/main/kotlin/de/brainsizzle/twitchbotpaint/paint/RotateAnimation.kt
package de.brainsizzle.twitchbotpaint.paint

class RotateAnimation(var degreesToRotatePerStep: Double, var stepsRemaining: Int) : Animation {

    override fun animate(shape: Shape): Boolean {
        if (stepsRemaining > 0) {
            shape.rotationDegress = shape.rotationDegress + degreesToRotatePerStep
            stepsRemaining--
            return false
        }
        return true
    }
}